command/oooo.go

```diff
@@ -1,21 +1,77 @@
 package command
 
+import (
+	"errors"
+	"strings"
+)
+
+func isVowel(char byte) bool {
+	switch char {
+	case 'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U':
+		return true
+	default:
+		return false
+	}
+}
+
 type OoooCommand struct{}
 
 func (oc *OoooCommand) Name() string {
 	return "oooo"
 }
 
+// OOOO-ify a provided string
+// - replace any "o"s if they exist
+// - if no "o"s exist, try to replace the vowel(s) closest to the center of the word
 func (oc *OoooCommand) Handle(message string) (string, error) {
 	oooo := ""
-	for _, ch := range message {
-		if ch == 'o' || ch == 'O' {
-			oooo += " OOOO "
+	if strings.Contains(strings.ToLower(message), "o") {
+		for _, ch := range message {
+			if ch == 'o' || ch == 'O' {
+				oooo += " OOOO "
+			} else {
+				oooo += string(ch)
+			}
+		}
+	} else {
+		messageLength := len(message)
+		midpoint := messageLength/2
+		vowelIndex := -1
+		if isVowel(message[midpoint]) {
+			vowelIndex = midpoint
 		} else {
-			oooo += string(ch)
+			// Search outwards to find the vowel closest to the center
+			for i:= 1; i < (messageLength - midpoint); i++ {
+				if beforeIndex := (midpoint - i); (beforeIndex) > 0 && isVowel(message[beforeIndex]) {
+					vowelIndex = beforeIndex
+					break
+				} else if afterIndex := (midpoint + i) ; isVowel(message[afterIndex]) {
+					vowelIndex = afterIndex
+					break
+				}
+			}
+		}
+		
+		if vowelIndex == -1 {
+		 	return "", errors.New("can't find any vowels to replace MMMM")
```

Comment:
Fun thought - what if instead of an error, we just return this as a string so it prints out to the chat?